app/src/main/java/org/mozilla/reference/browser/tabs/RemoteTabsActivity.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.mozilla.reference.browser.tabs
+
+import android.app.Activity
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseExpandableListAdapter
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_remotetabs.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.browser.storage.sync.TabEntry
+import mozilla.components.concept.sync.Device
+import mozilla.components.feature.tabs.TabsUseCases
+import org.mozilla.reference.browser.R
+import org.mozilla.reference.browser.ext.components
+import kotlin.random.Random
+
+
+class RemoteTabsActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_remotetabs)
+
+        val context = this
+        CoroutineScope(Dispatchers.Main).launch {
+
+            // TODO: we are only handling the optimistic case. Here are all the possible states:
+            // - No Sync account
+            // - Connected to FxA but not Sync (unreachable state on mobile at the moment).
+            // - Connected to Sync, but needs reconnection.
+            // - Connected to Sync, but tabs syncing disabled.
+            // - Connected to Sync, but tabs haven't been synced yet (they stay in memory after).
+            // - Connected to Sync, but only one device in the account (us), so no other tab to show.
+            // - Connected to Sync.
+
+            val remoteTabs = components.remoteTabs.getRemoteTabs()
+            val clientList = remoteTabs.map { (client, tabs) ->
+                ClientEntry(client, tabs.map { tab -> TabEntryWithID(tab.active()) })
+            }
+
+            val adapter = RemoteTabsListAdapter(context, clientList, components.useCases.tabsUseCases.addTab)
+            remote_tabs_list.setAdapter(adapter)
+            // Expand everything
+            for (i in 0 until adapter.groupCount) {
+                remote_tabs_list.expandGroup(i)
+            }
+        }
+    }
+}
+
+private data class ClientEntry(val device: Device, val tabs: List<TabEntryWithID>)
+
+data class TabEntryWithID(val entry: TabEntry, val id: Long = Random.nextLong())
+
+private class RemoteTabsListAdapter(val context: Context, val clients: List<ClientEntry>, val openTab: TabsUseCases.AddNewTabUseCase): BaseExpandableListAdapter() {
```

Comment:
Any reason to use a BaseExpandableListAdapter over just the regular RecyclerView?